presto-orc/src/main/java/com/facebook/presto/orc/OrcDataSource.java

```diff
@@ -20,6 +20,14 @@
 public interface OrcDataSource
         extends Closeable
 {
+    enum ReadType{
```

Comment:
- We probably need a new class `OrcReaderStats` and put this enum there. It's similar to `OrcWriterStats`
- Check how we implement `FlushReason` and do something similar to `ReadType`. Note that we use all capital + underscores for enum names.
- Also, follow how we track `FlushReason` and design probably for the reader. Expose counters with `@Managed` annotation.
-  Given you already have the orc stats exposed in the library, highly suggest implement this for Presto directly. I'm super interested in read pattern as well as we assume 90% reads go to footer for some of our low-latency workload. The tunneling is similar to `OrcWriterStats`, we can either create one in an injected class or bind it in Hive module.

Reply:
My apologies, will fix them and move the implementation to the OrcReaderStats and will update the PR.